ktor-client/ktor-client-okhttp/jvm/src/io/ktor/client/engine/okhttp/OkHttpWebsocketSession.kt

```diff
@@ -93,6 +93,14 @@ internal class OkHttpWebsocketSession(
             "${CloseReason.Codes.byCode(code.toShort())?.toString() ?: code}."))
     }
 
+    override fun onClosing(webSocket: WebSocket, code: Int, reason: String) {
+        super.onClosing(webSocket, code, reason)
+
+        _closeReason.complete(CloseReason(code.toShort(), reason))
+        outgoing.sendBlocking(Frame.Close(CloseReason(code.toShort(), reason)))
```

Comment:
Sending to a closed channel leads to an exception. Not sure what could happen when throwing it to okhttp

Reply:
I added an additional check here, but still not sure about what we shell do in such a case.